ksql-engine/src/test/java/io/confluent/ksql/topic/TopicDeleteInjectorTest.java

```diff
@@ -0,0 +1,201 @@
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.client.rest.exceptions.RestClientException;
+import io.confluent.ksql.metastore.MutableMetaStore;
+import io.confluent.ksql.metastore.model.StructuredDataSource;
+import io.confluent.ksql.parser.KsqlParser.PreparedStatement;
+import io.confluent.ksql.parser.tree.DropStream;
+import io.confluent.ksql.parser.tree.ListProperties;
+import io.confluent.ksql.parser.tree.QualifiedName;
+import io.confluent.ksql.parser.tree.Statement;
+import io.confluent.ksql.serde.DataSource.DataSourceSerDe;
+import io.confluent.ksql.services.FakeKafkaTopicClient;
+import io.confluent.ksql.statement.ConfiguredStatement;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.KsqlConstants;
+import io.confluent.ksql.util.KsqlException;
+import java.io.IOException;
+import java.util.Optional;
+import java.util.Set;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TopicDeleteInjectorTest {
+
+  @Rule
+  public final ExpectedException expectedException = ExpectedException.none();
+
+  @Mock
+  MutableMetaStore metaStore;
+  @Mock
+  StructuredDataSource<?> source;
+  @Mock
+  SchemaRegistryClient registryClient;
+
+  private FakeKafkaTopicClient topicClient;
+  private TopicDeleteInjector deleteInjector;
+
+  @Before
+  public void setUp() {
+    topicClient = new FakeKafkaTopicClient();
+    deleteInjector = new TopicDeleteInjector(metaStore, topicClient, registryClient);
+
+    when(metaStore.getSource("SOMETHING")).thenAnswer(inv -> source);
+    when(source.isSerdeFormat(DataSourceSerDe.AVRO)).thenReturn(true);
+  }
+
+  @Test
+  public void shouldDoNothingForNonDropStatements() {
+    // Given:
+    final ConfiguredStatement<ListProperties> listProperties =
+        givenStatement("LIST", new ListProperties(Optional.empty()));
+
+    // When:
+    final ConfiguredStatement<ListProperties> injected = deleteInjector.inject(listProperties);
+
+    // Then:
+    assertThat(injected, is(sameInstance(listProperties)));
+  }
+
+  @Test
+  public void shouldDoNothingIfNoDeleteTopic() {
+    // Given:
+    final ConfiguredStatement<DropStream> dropStatement = givenStatement(
+        "DROP SOMETHING", new DropStream(QualifiedName.of("something"), false, false));
+    final Set<String> originalTopics = ImmutableSet.copyOf(topicClient.listTopicNames());
+
+    // When:
+    final ConfiguredStatement<DropStream> injected = deleteInjector.inject(dropStatement);
+
+    // Then:
+    assertThat(injected, is(sameInstance(dropStatement)));
+    assertThat(topicClient.listTopicNames(), equalTo(originalTopics));
```

Comment:
I'd switch `topicClient ` to a mock, rather than a stub.  Makes for easier to read tests. e.g. this would be:

```suggestion
    verifyZeroInteractions(topicClient);
```

Reply:
It makes `givenTopic` way more complicated but I'll do it

Reply:
nevermind me... I actually don't need the given topic!